### ISSUE TITLE
Boost Filesystem fails to copy between volumes.  Consider replacing with std::filesystem.

### ISSUE THREAD

**Issue by jmpf** [Chaste/Chaste#100]
**Describe the bug**
Boost Filesystem fails to copy between volumes.  This is a known Boost bug: https://stackoverflow.com/questions/24209886/invalid-cross-device-link-error-with-boost-filesystem

**To Reproduce**
This may be specific to Boost 1.74.0 with /tmp and /home on different volumes:
```
jmpf@csu15256:~/Workspace/Chaste/build$ df /tmp/
Filesystem     1K-blocks      Used Available Use% Mounted on
/dev/nvme0n1p2 490617784 404090124  61532180  87% /
jmpf@csu15256:~/Workspace/Chaste/build$ df /home
Filesystem     1K-blocks      Used Available Use% Mounted on
/dev/nvme1n1p1 491133848 330703932 135408220  71% /home
```
```
ctest -VV -R TestFileFinder
...
13: Entering TestFindMatches
13: Passed
13: Entering TestCopying
13: 
13: /home/jmpf/Workspace/Chaste/global/test/TestFileFinder.hpp:362: Error: Test failed: 
13: Chaste error: ./global/src/FileFinder.cpp:332: boost::filesystem::copy_file: Invalid cross-device link: "/home/jmpf/Workspace/Chaste/global/test/TestFileFinder.hpp", "/tmp/jmpf/testoutput/TestFileFinder_TestCopying/TestFileFinder.hpp"
13: Failed
13: Entering TestDefaultConstructor
13: Passed
```

**Workaround**
Change environment variable to give a different output location
```
export CHASTE_TEST_OUTPUT="/home/jmpf/tmp-chaste/"
```

**Expected behaviour**
Test should pass

**System**
Please provide relevant details of your system, such as:
 * OS version: Ubuntu 22.04
 * Relevant Chaste dependency versions Boost: libboost-filesystem-dev:amd64              1.74.0.3ubuntu7

**Comment by jmpf** [comment]
We currently use -std=c++14 for compiling Chaste.

It's not clear if all the versions of C++ compilers in use will support the std::filesystem.  Probably they all will but some may need `#include <experimental/filesystem>` rather than  `#include <filesystem>` .

```
jmpf@csu15256:~$ cat temp.cc
#include <filesystem>
namespace fs=std::filesystem;

int main(){}
jmpf@csu15256:~$ g++ -std=c++17 temp.cc 
jmpf@csu15256:~$ g++ -std=c++14 temp.cc 
temp.cc:2:19: error: ‘filesystem’ is not a namespace-name
    2 | namespace fs=std::filesystem;
      |                   ^~~~~~~~~~
```

**Comment by mirams** [comment]
Any problem with going straight to compiling with c++17 ? I've forgotten if we have a 'policy' on that move @fcooper8472 ?

**Comment by jmpf** [comment]
@mirams How many of our users are using C++ compilers that cannot handle C++17?

**Comment by mirams** [comment]
our what?

**Comment by jmpf** [comment]
See also #107 -- There's now a compelling reason to switch on C++17 features.

**Comment by jmpf** [comment]
Will make https://github.com/Chaste/Chaste/tree/issue_100 into pull request when we're green

**Comment by jmpf** [comment]
Fixed by #115
